packages/data-point/lib/resolve/reducer-function.js

```diff
@@ -81,16 +86,37 @@ function resolve (filterStore, accumulator, reducerFunction) {
         if (err) {
           return reject(err)
         }
+
+        // if the result is a resolved value then return the accumulator with that value and a resolvedValue
+        if (value.isResolved) {
+          return utils.assign(currentAccumulator, {
+            value: value.value,
+            resolvedValue: value.value
+          })
+        }
+
         resolve(utils.set(currentAccumulator, 'value', value))
       })
     })
   }
 
   // callbackFunction is assumed to be either sync
   // or Promise returned value
-  return Promise.try(callbackFunction.bind(null, accumulator)).then(result => {
-    return utils.set(currentAccumulator, 'value', result)
-  })
+  return Promise.try(callbackFunction.bind(null, accumulator))
+    .then(result => {
+      // if the result is a resolved value then return the accumulator with that value and a resolvedValue
+      if (result.isResolved) {
+        return utils.assign(currentAccumulator, {
+          value: result.value,
+          resolvedValue: result.value
+        })
+      }
+
+      return utils.set(currentAccumulator, 'value', result)
+    })
+    .catch(err => {
```

Comment:
This `catch` wasn't here before (and isn't really doing anything). If we want to remove the `catch` or throw an error or something else please LMK.